### ISSUE TITLE
"Right Click -> Copy Remote File URL" Does not add selected line numbers to URL with Stash/Bitbucket 

### ISSUE THREAD

**Issue by rockboynton** [gitkraken/vscode-gitlens#1798]
- GitLens Version: 11.6.1
- Git Version: Run 2.25.1

- VSCode Version: Run 1.63.2
- OS Version: Windows 10 using remote SSH extension into Ubuntu 20.04.3 LTS machine

Steps to Reproduce:
1. Highlight lines in a file
2. "Right Click on file name -> Open File on Remote" 
3. See the link that opens contains the selected line numbers in the URL
4. Try "Right Click -> Copy Remote File URL" and paste it
5. See that link does not have the line numbers.

[see here for original ticket](https://github.com/microsoft/vscode/issues/137249#issue-1054035874)

**Comment by danielyan86129** [comment]
Hi @rockboynton would you please share your remote provider config? I can't even get it to work with Stash at all. Thx!

**Comment by rockboynton** [comment]
We use a locally hosted Stash server, can you clarify what you mean by "provider config"? I do see in the `settings.json` there is this:
```json
"gitlens.remotes": [
        {
            "domain": "our stash server url",
            "type": "BitbucketServer"
        }
    ],
```
Hopefully that helps!

**Comment by danielyan86129** [comment]
@rockboynton yes that's what I mean. Basically I have an issue setting it up: https://github.com/gitkraken/vscode-gitlens/issues/1796
Have you not found a need to set it up as a `"type": "Custom"`?

**Comment by hriekehof** [comment]
I have the problem when i use `Copy Remote File URL From`. The link does not contain the line numbers. When i use `Copy Remote File URL` it has the line numbers but then my local branch sha is used which it not existent on remote. Would be nice to have an option to always use main or to configure a default branch ;)

**Comment by rockboynton** [comment]
> I have the problem when i use `Copy Remote File URL From`. But for me the type is `Github`. The link does not contain the line numbers. When i use `Copy Remote File URL` it has the line numbers but then my local branch sha is used which it not existent on remote. Would be nice to have an option to always use main or to configure a default branch ;)

Yeah my org switched to GitHub and we still have this issue. Quite annoying
